modules/materials-page/materials-page.service.js

```diff
@@ -0,0 +1,80 @@
+const Materials = require('./materials-page.model');
+const {
+  MATERIAL_NOT_FOUND,
+} = require('../../error-messages/material.messages');
+const {
+  STATUS_CODES: { NOT_FOUND },
+} = require('../../consts/status-codes');
+const RuleError = require('../../errors/rule.error');
+
+const createTranslations = require('../../utils/createTranslations');
+const {
+  addTranslations,
+  updateTranslations,
+  deleteTranslations,
+} = require('../translations/translations.service');
+
+class MaterialsService {
+  async getAllMaterialsBlocks() {
+    const items = await Materials.find({}).exec();
+    const count = items.length;
+
+    return {
+      items,
+      count,
+    };
+  }
+
+  async getMaterialsBlockById(id) {
+    const materialsBlock = await Materials.findById(id).exec();
+
+    if (!materialsBlock) throw new RuleError(MATERIAL_NOT_FOUND, NOT_FOUND);
+
+    return materialsBlock;
+  }
+
+  async addMaterialsBlock(materialsBlock) {
+    materialsBlock.translationsKey = await addTranslations(
+      createTranslations(materialsBlock)
+    );
+
+    return new Materials(materialsBlock).save();
+  }
+
+  async deleteMaterialsBlock(id) {
+    const materialsBlock = await Materials.findByIdAndDelete(id).exec();
+
+    if (materialsBlock) {
+      await deleteTranslations(materialsBlock.translationsKey);
+
+      return materialsBlock;
+    }
+    throw new RuleError(MATERIAL_NOT_FOUND, NOT_FOUND);
+  }
+
+  async updateMaterialsBlock(id, materialsBlock) {
+    const foundMaterialsBlock = await Materials.findById(id).exec();
+
+    if (foundMaterialsBlock) {
+      await updateTranslations(
+        foundMaterialsBlock.translationsKey,
+        createTranslations(materialsBlock)
+      );
+    } else {
+      throw new RuleError(MATERIAL_NOT_FOUND, NOT_FOUND);
+    }
+
+    const updatedMaterialsBlock = await Materials.findByIdAndUpdate(
+      id,
+      materialsBlock,
+      { new: true }
+    ).exec();
+
+    if (updatedMaterialsBlock) {
+      return updatedMaterialsBlock;
+    }
+    return false;
```

Comment:
why do you need return false here? mb better throw error if something went wrong